parsing/dialects/intermediate/dialects/dialect1/src/test/java/org/apache/calcite/test/Dialect1ValidatorTest.java

```diff
@@ -28,4 +38,15 @@
         SqlTestFactory.INSTANCE
             .with("parserFactory", Dialect1ParserImpl.FACTORY));
   }
+
+  @Test public void testCaretNegation() throws SqlParseException {
+    String sql = "select * from dept where ^deptno = 1";
+    SqlParser.Config config = configBuilder()
+        .setParserFactory(Dialect1ParserImpl.FACTORY)
+        .build();
+    SqlParser sqlParserReader = SqlParser.create(new StringReader(sql), config);
+    SqlNode node = sqlParserReader.parseQuery();
+    SqlValidator validator = tester.getValidator();
+    validator.validate(node);
```

Comment:
Would it be possible to call toSqlString() on the validated node similar to what is done [here](https://github.com/googleinterns/calcite/blob/aefe2b15dc4f187a740d2c4227d0f642320b6153/core/src/test/java/org/apache/calcite/sql/test/AbstractSqlTester.java#L505), and assert that it rewrites to what should be expected? Otherwise, the ^ could theoretically be dropped and we wouldn't know.

Reply:
Completely agree. @dstekol are there benefits of using toSqlString() over rewritesTo(), as is done in the link below?

https://github.com/googleinterns/calcite/blob/aefe2b15dc4f187a740d2c4227d0f642320b6153/parsing/dialects/intermediate/dialects/dialect1/src/test/java/org/apache/calcite/test/Dialect1ValidatorTest.java#L42

Reply:
Michael mentioned in the design doc that ^ gets dropped by the toSql() call of the testing framework (since it usually gets used for annotating errors), so this is a workaround.

Reply:
I think that is also why this particular test sets everything up from scratch. We likely won't need to do this for anything else, but this is a special case (at least until someone figures out a way to indicate to the tester than ^ is being used as an operator).

Reply:
Yeah I initially set it up like that because the caret gets dropped, but I I think calling getTester().rewritesTo() would be a lot cleaner and that shouldn't remove the caret operator. If rewritesTo() is called on a sql() object the operator is removed.

Reply:
Yeah I initially set it up like that because the caret gets dropped, but I I think calling getTester().checkRewrite() which calls the toSqlString() method could be a lot cleaner and that shouldn't remove the caret operator. If rewritesTo() is called on a sql() object the operator is removed.

Reply:
Ah I see. Let's at least comment that then.

Reply:
I updated the test case and left a comment. Thanks for pointing this out @dasch-1 and @dstekol.